src/core/string_map.cc

```diff
@@ -101,6 +101,74 @@ sds StringMap::Find(std::string_view key) {
   return GetValue(str);
 }
 
+void StringMap::RandomPair(sds* key, sds* value) {
+  auto itr = begin();
+  for (unsigned int i = 0; i < (rand() % Size()); ++i)
+    ++itr;
+  *key = itr->first;
+  if (value != nullptr)
+    *value = itr->second;
+  return;
+}
+
+unsigned int StringMap::RandomPairsUnique(unsigned int count, sds* keys, sds* vals) {
+  unsigned int total_size = Size();
```

Comment:
Same here, lets return a vector<pair<sds,sds>>. And then you don't need create raw arrays manually or the repetitive `if (!with_vales.empty()) clause`

Reply:
@dranikpg I prefer only return vals upon request. The count value theoretically can be large, and the value itself could be large. Unlike RandomPair which only returns one pair, this function potentially could significantly increase memory consumption if we always return everything.

Reply:
the function now takes in two vecotor references (keys, vals), and a boolean value indicating whether the `vals` needs to be filled in.

Reply:
Then just return a std::pairt<vector, vector>, the second one will be empty and therefore no memory will wasted for an empty vector other than the pointer + size in the second field of pair

Reply:
@kostasrim please take a look at the current implementation. I think it is a personal choice to use two vectors separately or a pair of vectors...

Reply:
I think it doesn't make much sense to have the two vectors as input parameters. If you do what @dranikpg suggested and return them instead, we could squash that function into one. Furthermore, I see no reason why from an API pov I would want to pass the result vectors as input parameters for this particular contexts. It's just noise.